docs/frontend/styles.md

```diff
@@ -15,11 +15,35 @@ Important files when working with SASS in the project:
 - variables:
   [/app/assets/stylesheets/variables.scss](https://github.com/thepracticaldev/dev.to/blob/master/app/assets/stylesheets/variables.scss)
 - mixins:
-  [/app/assets/stylesheets/_mixins.scss](https://github.com/thepracticaldev/dev.to/blob/master/app/assets/stylesheets/_mixins.scss)
+  [/app/assets/stylesheets/\_mixins.scss](https://github.com/thepracticaldev/dev.to/blob/master/app/assets/stylesheets/_mixins.scss)
 
 SASS is compiled and served using
 [Sprockets](https://github.com/rails/sprockets-rails) which packages static
 assets in Rails.
 
 For more about branding, theming or design in general in regards to DEV, refer
 to the [Design Guide](/design) documentation.
+
+# Significant "shell" CSS and HTML changes and deployment
+
+We use serviceworkers to cache the base application shell in user browsers so
+that we only have to serve partial pages.
+
+This means that when we make _substantial_ updates to core styles that affect
+all pages or affect the home page in a big way, we need to increment the
+`impactful_shell_version` number in `ApplicationHelper`. Small adjustments, even
+if they result in slight design regressions do not need this number incremented
+because as soon as users visit the site they will get a fresh shell download in
+the background.
```

Comment:
Also think this would be helpful in clarifying the paragraph below, which reads: 
> The key is to have good communication around what may and may not require a a bump to the `impactful_shell_version`.